### PR TITLE
Additional audio API methods

### PR DESCRIPTION
**Pull request by ejb**
This (finally!) implements #36

* getStatus
* getVolume / setVolume
* getDuration

### CODE REVIEW

File: packages/replay-web/src/device.ts [hunk 1]
```diff
@@ -46,6 +47,7 @@ export function getAudio(
       );
     }
     const { buffer, playState } = data;
+    const gainNode = playState?.gainNode || audioContext.createGain();
```
Comment (edbentley) [inline packages/replay-web/src/device.ts:50]: I'm worried that the gainNode may not get cleaned up or conflict with other sounds. How about you create a new one every time `play` is called, and then access it in `setVolume` etc through `playState`?

Comment (ejb) [reply packages/replay-web/src/device.ts]: Done!

**Comment by ejb** [comment]
Thanks for your feedback, I think I've fixed the issues we discussed.

**Review by edbentley — CHANGES_REQUESTED** [review]
The code looks great, but unfortunately this still isn't working for me:

```js
    const sound = device.audio(fileName);
    sound.play();
    sound.setVolume(0.1);
```

**Comment by edbentley** [comment]
Actually I believe this is an existing bug with `playState` since `getPosition()` also has this issue. Just adding these new APIs has made it more visible!

**Comment by ejb** [comment]
Ok, think I've resolved those issues. Seemed to be caused by the value of `playState` being cached in those instances. Reading from `data.playState` instead gives correct values.

**Review by edbentley — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**